### ISSUE TITLE
semi: [2, "never"] not erroring when semi-colon is after a static property

### ISSUE THREAD

**Issue by jeffshaver** [babel/babel-eslint#260]
I originally filed this issue on the original eslint repo, but I was told that I should be filing it here (eslint/eslint#5366)

ESLint Version: 2.1.0
Babel ESLint Version: 5.0.0

If I have `semi: [2, "never"]` set, I would imagine this would cause an error:

```javascript

export class MyClass extends Component {
  static propTypes = {
  }; // <-- expected this to give an error

  render () {
    // code
  }
}
```

Currently, I am getting 0 output from ESLint about this.

* [Actual source code](https://github.com/jeffshaver/safe-framework/blob/scatter-plot-component/src/DefaultScatterPlot/DefaultScatterPlot.jsx#L13)

* [ESLint Config](https://github.com/jeffshaver/safe-framework/blob/scatter-plot-component/.eslintrc)

**Comment by hzoo** [comment]
It's not going to error because the rule itself doesn't know about class properties (it's not in https://github.com/eslint/eslint/blob/master/lib/rules/semi.js#L160-L179).

We would have to fork the rule in https://github.com/babel/eslint-plugin-babel to include class properties and use that rule instead.

**Comment by jeffshaver** [comment]
@hzoo should i request this change in eslint-plugin-babel?

**Comment by hzoo** [comment]
Yeah although it's basically the opposite of https://github.com/babel/eslint-plugin-babel/issues/43 so I think it's the same issue - you can post a comment there

Basically semi needs to support class properties in general

**Comment by xgqfrms-gildata** [comment]
# `1` for `warnings` and `2` for `errors`

http://eslint.org/docs/developer-guide/working-with-custom-formatters#the-message-object

# `"semi": 2`

```js
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "rules": {
        "semi": 2
    }
}
```

http://eslint.org/docs/user-guide/configuring

http://eslint.org/docs/rules/no-compare-neg-zero

```js
"semi": [2, "always"]
```

**Comment by xgqfrms-gildata** [comment]
```js
"semi": [2, "always"],

```

http://eslint.org/docs/rules/no-compare-neg-zero
